// spaces-client/lib/blog.js
// Generated by CoffeeScript 1.6.3
(function() {
  var Blog, exports;

  Blog = (function() {
    var Faker, Resource;

    function Blog() {}

    Faker = require('faker');

    Resource = require('./resource');

    Blog.createBlog = function(site, sessionId, parentId, userId, onsuccess, onfail, oncreate) {
      var data;
      data = {
        blog: {
          title: Faker.Lorem.sentences(1),
          body: Faker.Lorem.sentences(4)
        }
      };
      if (parentId) {
        data['blog']['parent_id'] = parentId;
      }
      return Resource.createResource(site, userId, sessionId, 'blog', data, onsuccess, onfail, oncreate);
    };

    Blog.updateBlog = function(site, sessionId, refObjId, userId, onsuccess, onfail) {
      var data;
      data = {
        blog: {
          title: Faker.Lorem.sentences(1)
        }
      };
      return Resource.editResource(site, userId, sessionId, 'blog', refObjId, data, onsuccess, onfail);
    };

    Blog.deleteBlog = function(site, sessionId, refObjId, userId, onsuccess, onfail, ondestroy) {
      return Resource.deleteResource(site, userId, sessionId, refObjId, 'blog', onsuccess, onfail, ondestroy);
    };

    Blog.createBlogpost = function(site, sessionId, blogId, userId, onsuccess, onfail, oncreate) {
      var data;
      data = {
        blog_post: {
          title: Faker.Lorem.sentences(1),
          body: Faker.Lorem.sentences(20),
          parent_id: blogId
        }
      };
      return Resource.createResource(site, userId, sessionId, 'blog_post', data, onsuccess, onfail, oncreate);
    };

    Blog.updateBlogpost = function(site, sessionId, refObjId, userId, onsuccess, onfail) {
      var data;
      data = {
        blog_post: {
          body: Faker.Lorem.sentences(20)
        }
      };
      return Resource.editResource(site, userId, sessionId, 'blog_post', refObjId, data, onsuccess, onfail);
    };

    Blog.deleteBlogpost = function(site, sessionId, refObjId, userId, onsuccess, onfail, ondestroy) {
      return Resource.deleteResource(site, userId, sessionId, refObjId, 'blog_post', onsuccess, onfail, ondestroy);
    };

    return Blog;

  })();

  module.exports = exports = Blog;

}).call(this);
