// spaces-client/lib/blog.js
// Generated by CoffeeScript 1.6.3
(function() {
  var Blog, exports;

  Blog = (function() {
    var Faker, Resource;

    function Blog() {}

    Faker = require('faker');

    Resource = require('./resource');

    Blog.showBlog = function(site, sessionId, resourceId, refObjId, userId, onsuccess, onfail) {
      var addlHeaders;
      addlHeaders = [];
      return Resource.getResource(site.full_url, userId, sessionId, 'blogs/' + resourceId, addlHeaders, (function(res) {
        return onsuccess();
      }), (function() {
        return onfail();
      }));
    };

    Blog.createBlog = function(site, sessionId, resourceId, parentId, userId, onsuccess, onfail, oncreate) {
      var data;
      data = {
        blog: {
          title: Faker.Lorem.sentences(1),
          body: Faker.Lorem.sentences(4)
        }
      };
      if (parentId) {
        data['blog']['parent_id'] = parentId;
      }
      return Resource.createResource(site, userId, sessionId, 'blogs', data, onsuccess, onfail, oncreate);
    };

    Blog.updateBlog = function(site, sessionId, resourceId, parentId, userId, onsuccess, onfail) {
      var data;
      data = {
        blog: {
          title: Faker.Lorem.sentences(1)
        }
      };
      return Resource.editResource(site, userId, sessionId, 'blogs/' + resourceId, data, onsuccess, onfail);
    };

    Blog.deleteBlog = function(site, sessionId, resourceId, userId, onsuccess, onfail, ondestroy) {
      return Resource.deleteResource(site, userId, sessionId, 'blogs/' + resourceId, onsuccess, onfail, ondestroy);
    };

    Blog.createBlogpost = function(site, sessionId, resourceId, parentId, userId, onsuccess, onfail, oncreate) {
      var data;
      data = {
        blog_post: {
          title: Faker.Lorem.sentences(1),
          body: Faker.Lorem.sentences(20),
          parent_id: parentId
        }
      };
      return Resource.createResource(site, userId, sessionId, 'blog_posts', data, onsuccess, onfail, oncreate);
    };

    Blog.updateBlogpost = function(site, sessionId, resourceId, parentId, userId, onsuccess, onfail) {
      var data;
      data = {
        blog_post: {
          body: Faker.Lorem.sentences(20)
        }
      };
      return Resource.editResource(site, userId, sessionId, 'blog_posts/' + resourceId, data, onsuccess, onfail);
    };

    Blog.deleteBlogpost = function(site, sessionId, resourceId, parentId, userId, onsuccess, onfail, ondestroy) {
      return Resource.deleteResource(site, userId, sessionId, 'blog_posts/' + resourceId, onsuccess, onfail, ondestroy);
    };

    return Blog;

  })();

  module.exports = exports = Blog;

}).call(this);
