// spaces-client/lib/forum.js
// Generated by CoffeeScript 1.6.3
(function() {
  var $, Faker, Forum, Resource, exports;

  Faker = require('faker');

  $ = require('jquery');

  Resource = require('./resource.js');

  Forum = (function() {
    function Forum() {}

    Forum.create = function(site, parentId, userId) {
      var data;
      data = {
        discussion_archive: {
          name: Faker.Lorem.sentences(1),
          description: Faker.Lorem.sentences(4)
        }
      };
      if (parentId) {
        data['discussion_archive']['parent_id'] = parentId;
      }
      return Resource.createResource(site, userId, 'discussion_archive', data);
    };

    Forum.updateForum = function(site, parentId, userId) {
      var data;
      data = {
        discussion_archive: {
          description: Faker.Lorem.sentences(4)
        }
      };
      return Resource.editResource(site, userId, 'discussion_archive', data);
    };

    Forum.deleteForum = function(site, parentId, userId) {
      return Resource.deleteResource(site, userId, 'discussion_archive');
    };

    Forum.createDiscussion = function(site, forumId, userId) {
      var data;
      data = {
        discussion: {
          title: Faker.Lorem.sentences(1),
          body: Faker.Lorem.sentences(4)
        }
      };
      if (forumId) {
        data['discussion']['parent_id'] = forumId;
      }
      return Resource.createResource(site, userId, 'discussion', data);
    };

    Forum.updateDiscussion = function(site, forumId, userId) {
      var data;
      data = {
        discussion: {
          body: Faker.Lorem.sentences(4)
        }
      };
      return Resource.editResource(site, userId, 'discussion', data);
    };

    Forum.deleteDiscussion = function(site, parentId, userId) {
      return Resource.deleteResource(site, userId, 'discussion');
    };

    return Forum;

  })();

  module.exports = exports = Forum;

}).call(this);
