// spaces-client/lib/forum.js
// Generated by CoffeeScript 1.6.3
(function() {
  var Forum, exports;

  Forum = (function() {
    var Faker, Resource;

    function Forum() {}

    Faker = require('faker');

    Resource = require('./resource');

    Forum.createForum = function(site, sessionId, resourceId, parentId, userId, onsuccess, onfail, oncreate) {
      var data;
      data = {
        discussion_archive: {
          name: Faker.Lorem.sentences(1),
          description: Faker.Lorem.sentences(4)
        }
      };
      if (parentId) {
        data['discussion_archive']['parent_id'] = parentId;
      }
      return Resource.createResource(site, userId, sessionId, 'discussion_archives', data, onsuccess, onfail, oncreate);
    };

    Forum.showForum = function(site, sessionId, resourceId, parentId, userId, onsuccess, onfail) {
      var addlHeaders;
      addlHeaders = {
        "Cookie": ["_social_navigator_session=" + sessionId, "path=/"]
      };
      return Resource.getResource(site, userId, sessionId, 'discussion_archives/' + resourceId, addlHeaders, (function(res) {
        return onsuccess();
      }), (function() {
        return onfail();
      }));
    };

    Forum.updateForum = function(site, sessionId, resourceId, parentId, userId, onsuccess, onfail) {
      var data;
      data = {
        discussion_archive: {
          description: Faker.Lorem.sentences(4)
        }
      };
      return Resource.editResource(site, userId, sessionId, 'discussion_archives/' + resourceId, data, onsuccess, onfail);
    };

    Forum.deleteForum = function(site, sessionId, resourceId, parentId, userId, onsuccess, onfail, ondestroy) {
      return Resource.deleteResource(site, userId, sessionId, 'discussion_archives/' + resourceId, onsuccess, onfail, ondestroy);
    };

    Forum.createDiscussion = function(site, sessionId, resourceId, parentId, userId, onsuccess, onfail, oncreate) {
      var data;
      data = {
        discussion: {
          title: Faker.Lorem.sentences(1),
          body: Faker.Lorem.sentences(4),
          parent_id: parentId
        }
      };
      return Resource.createResource(site, userId, sessionId, 'discussions', data, onsuccess, onfail, oncreate);
    };

    Forum.showDiscussion = function(site, sessionId, resourceId, parentId, userId, onsuccess, onfail) {
      var addlHeaders;
      addlHeaders = {
        "Cookie": ["_social_navigator_session=" + sessionId, "path=/"]
      };
      return Resource.getResource(site, userId, sessionId, 'discussions/' + resourceId, addlHeaders, (function(res) {
        return onsuccess();
      }), (function() {
        return onfail();
      }));
    };

    Forum.updateDiscussion = function(site, sessionId, resourceId, parentId, userId, onsuccess, onfail) {
      var data;
      data = {
        discussion: {
          body: Faker.Lorem.sentences(4)
        }
      };
      return Resource.editResource(site, userId, sessionId, 'discussions/' + resourceId, data, onsuccess, onfail);
    };

    Forum.deleteDiscussion = function(site, sessionId, resourceId, parentId, userId, onsuccess, onfail, ondestroy) {
      return Resource.deleteResource(site, userId, sessionId, 'discussions/' + resourceId, onsuccess, onfail, ondestroy);
    };

    return Forum;

  })();

  module.exports = exports = Forum;

}).call(this);
