// spaces-client/lib/ideastorm.js
// Generated by CoffeeScript 1.6.3
(function() {
  var Ideastorm, exports;

  Ideastorm = (function() {
    var Faker, Resource;

    function Ideastorm() {}

    Faker = require('faker');

    Resource = require('./resource');

    Ideastorm.createIdeastorm = function(site, sessionId, parentId, userId, onsuccess, onfail, oncreate) {
      var data;
      data = {
        ideastorm: {
          name: Faker.Lorem.sentences(1),
          description: Faker.Lorem.sentences(4),
          close_at: new Date().toISOString()
        }
      };
      if (parentId) {
        data['ideastorm']['parent_id'] = parentId;
      }
      return Resource.createResource(site, userId, sessionId, 'ideastorm', data, onsuccess, onfail, oncreate);
    };

    Ideastorm.updateIdeastorm = function(site, sessionId, resourceId, userId, onsuccess, onfail) {
      var data;
      data = {
        ideastorm: {
          description: Faker.Lorem.sentences(4)
        }
      };
      return Resource.editResource(site, userId, sessionId, 'ideastorm', resourceId, data, onsuccess, onfail);
    };

    Ideastorm.deleteIdeastorm = function(site, sessionId, resourceId, userId, onsuccess, onfail, ondestroy) {
      return Resource.deleteResource(site, userId, sessionId, resourceId, 'ideastorm', onsuccess, onfail, ondestroy);
    };

    Ideastorm.createIdea = function(site, sessionId, ideastormId, userId, onsuccess, onfail, oncreate) {
      var data;
      data = {
        idea: {
          title: Faker.Lorem.sentences(1),
          body: Faker.Lorem.sentences(4),
          parent_id: ideastormId
        }
      };
      return Resource.createResource(site, userId, sessionId, 'idea', data, onsuccess, onfail, oncreate);
    };

    Ideastorm.updateIdea = function(site, sessionId, resourceId, userId, onsuccess, onfail) {
      var data;
      data = {
        idea: {
          body: Faker.Lorem.sentences(4)
        }
      };
      return Resource.editResource(site, userId, sessionId, 'idea', resourceId, data, onsuccess, onfail);
    };

    Ideastorm.deleteIdea = function(site, sessionId, resourceId, userId, onsuccess, onfail, ondestroy) {
      return Resource.deleteResource(site, userId, sessionId, resourceId, 'idea', onsuccess, onfail, ondestroy);
    };

    return Ideastorm;

  })();

  module.exports = exports = Ideastorm;

}).call(this);
