// spaces-client/lib/ideastorm.js
// Generated by CoffeeScript 1.6.3
(function() {
  var $, Faker, Ideastorm, Resource, exports;

  Faker = require('faker');

  $ = require('jquery');

  Resource = require('./resource.js');

  Ideastorm = (function() {
    function Ideastorm() {}

    Ideastorm.create = function(site, parentId, userId) {
      var data;
      data = {
        ideastorm: {
          name: Faker.Lorem.sentences(1),
          description: Faker.Lorem.sentences(4)
        }
      };
      if (parentId) {
        data['ideastorm']['parent_id'] = parentId;
      }
      return Resource.createResource(site, userId, 'ideastorm', data);
    };

    Ideastorm.updateIdeastorm = function(site, parentId, userId) {
      var data;
      data = {
        ideastorm: {
          description: Faker.Lorem.sentences(4)
        }
      };
      return Resource.editResource(site, userId, 'ideastorm', data);
    };

    Ideastorm.deleteIdeastorm = function(site, parentId, userId) {
      return Resource.deleteResource(site, userId, 'ideastorm');
    };

    Ideastorm.createIdea = function(site, ideastormId, userId) {
      var data;
      data = {
        idea: {
          title: Faker.Lorem.sentences(1),
          body: Faker.Lorem.sentences(4)
        }
      };
      if (ideastormId) {
        data['idea']['parent_id'] = ideastormId;
      }
      return Resource.createResource(site, userId, 'idea', data);
    };

    Ideastorm.updateIdea = function(site, ideastormId, userId) {
      var data;
      data = {
        idea: {
          body: Faker.Lorem.sentences(4)
        }
      };
      return Resource.editResource(site, userId, 'idea', data);
    };

    Ideastorm.deleteIdea = function(site, ideastormId, userId) {
      return Resource.deleteResource(site, userId, 'idea');
    };

    return Ideastorm;

  })();

  module.exports = exports = Ideastorm;

}).call(this);
