// spaces-client/lib/ideastorm.js
// Generated by CoffeeScript 1.6.3
(function() {
  var Ideastorm, exports;

  Ideastorm = (function() {
    var Faker, Resource;

    function Ideastorm() {}

    Faker = require('faker');

    Resource = require('./resource');

    Ideastorm.createIdeastorm = function(site, sessionId, resourceId, parentId, userId, onsuccess, onfail, oncreate) {
      var data;
      data = {
        ideastorm: {
          name: Faker.Lorem.sentences(1),
          description: Faker.Lorem.sentences(4),
          close_at: new Date().toISOString()
        }
      };
      if (parentId) {
        data['ideastorm']['parent_id'] = parentId;
      }
      return Resource.createResource(site, userId, sessionId, 'ideastorms', data, onsuccess, onfail, oncreate);
    };

    Ideastorm.showIdeastorm = function(site, sessionId, resourceId, parentId, userId, onsuccess, onfail) {
      var addlHeaders;
      addlHeaders = {
        "Accept": "application/vnd.moxiesoft.spaces-v1+json",
        "Cookie": ["_social_navigator_session=" + sessionId, "path=/"]
      };
      return Resource.getResource(site, userId, sessionId, 'ideastorms/' + resourceId, addlHeaders, onsuccess, onfail);
    };

    Ideastorm.updateIdeastorm = function(site, sessionId, resourceId, parentId, userId, onsuccess, onfail) {
      var data;
      data = {
        ideastorm: {
          description: Faker.Lorem.sentences(4)
        }
      };
      return Resource.editResource(site, userId, sessionId, 'ideastorms/' + resourceId, data, onsuccess, onfail);
    };

    Ideastorm.deleteIdeastorm = function(site, sessionId, resourceId, parentId, userId, onsuccess, onfail, ondestroy) {
      return Resource.deleteResource(site, userId, sessionId, 'ideastorms/' + resourceId, onsuccess, onfail, ondestroy);
    };

    Ideastorm.createIdea = function(site, sessionId, resourceId, parentId, userId, onsuccess, onfail, oncreate) {
      var data;
      data = {
        idea: {
          title: Faker.Lorem.sentences(1),
          body: Faker.Lorem.sentences(4),
          parent_id: parentId
        }
      };
      return Resource.createResource(site, userId, sessionId, 'ideas', data, onsuccess, onfail, oncreate);
    };

    Ideastorm.showIdea = function(site, sessionId, resourceId, parentId, userId, onsuccess, onfail) {
      var addlHeaders;
      addlHeaders = {
        "Accept": "application/vnd.moxiesoft.spaces-v1+json",
        "Cookie": ["_social_navigator_session=" + sessionId, "path=/"]
      };
      return Resource.getResource(site, userId, sessionId, 'ideas/' + resourceId, addlHeaders, onsuccess, onfail);
    };

    Ideastorm.updateIdea = function(site, sessionId, resourceId, parentId, userId, onsuccess, onfail) {
      var data;
      data = {
        idea: {
          body: Faker.Lorem.sentences(4)
        }
      };
      return Resource.editResource(site, userId, sessionId, 'ideas/' + resourceId, data, onsuccess, onfail);
    };

    Ideastorm.deleteIdea = function(site, sessionId, resourceId, parentId, userId, onsuccess, onfail, ondestroy) {
      return Resource.deleteResource(site, userId, sessionId, 'ideas/' + resourceId, onsuccess, onfail, ondestroy);
    };

    return Ideastorm;

  })();

  module.exports = exports = Ideastorm;

}).call(this);
