// spaces-client/lib/spaces-client.js
// Generated by CoffeeScript 1.6.3
(function() {
  var SpacesClient;

  SpacesClient = (function() {
    function SpacesClient() {}

    SpacesClient.Activity = require("./activity.js");

    SpacesClient.Blog = require("./blog.js");

    SpacesClient.Category = require("./category.js");

    SpacesClient.Folder = require("./folder.js");

    SpacesClient.Group = require("./group.js");

    SpacesClient.Session = require("./session.js");

    SpacesClient.Site = require("./site.js");

    SpacesClient.User = require("./user.js");

    SpacesClient.logger = null;

    SpacesClient.setLogger = function(logger) {
      return this.logger = logger;
    };

    SpacesClient.mothershipUrl = null;

    SpacesClient.setMothershipURL = function(url) {
      return this.mothershipUrl = url;
    };

    SpacesClient.apiKey = null;

    SpacesClient.setApiKey = function(key) {
      return this.apiKey = key;
    };

    return SpacesClient;

  })();

  module.exports = SpacesClient;

  module.exports.setLogger = SpacesClient.setLogger;

  module.exports.getLogger = SpacesClient.getLogger;

}).call(this);
