// spaces-client/lib/user.js
// Generated by CoffeeScript 1.6.3
(function() {
  var UserAPI, exports, logger;

  logger = require('./logger.js');

  UserAPI = (function() {
    function UserAPI() {}

    UserAPI.create = function(data, host, onsuccess, onfail) {
      var $, Spaces;
      Spaces = require('./spaces-client.js');
      $ = require("jquery");
      return $.ajax(host + '/users.json', {
        type: 'POST',
        dataType: 'json',
        data: JSON.stringify(data),
        beforeSend: function(request) {
          request.setRequestHeader("Accept", "application/vnd.moxiesoft.spaces-v1+json");
          return request.setRequestHeader("Content-Type", "application/json");
        },
        error: function(jqXHR, textStatus, errorThrown) {
          logger.error("[%s][No user] User.create failed: %s - %s - %j", host, textStatus, errorThrown, jqXHR);
          return onfail(errorThrown);
        },
        success: function(data) {
          return onsuccess(data);
        }
      });
    };

    return UserAPI;

  })();

  module.exports = exports = UserAPI;

}).call(this);
