// spaces-client/lib/user.js
// Generated by CoffeeScript 1.6.3
(function() {
  var UserAPI, exports;

  UserAPI = (function() {
    function UserAPI() {}

    UserAPI.create = function(data, host, onsuccess, onfail) {
      var $;
      $ = require("jquery");
      return $.ajax(host + '/users.json', {
        type: 'POST',
        dataType: 'json',
        data: JSON.stringify(data),
        beforeSend: function(request) {
          request.setRequestHeader("Accept", "application/vnd.moxiesoft.mothership-v1+json");
          return request.setRequestHeader("Content-Type", "application/json");
        },
        error: function(jqXHR, textStatus, errorThrown) {
          console.log("Error creating user: " + textStatus + " - " + errorThrown + " - " + JSON.stringify(jqXHR));
          return onfail(errorThrown);
        },
        success: function(data) {
          console.log("User Created: " + JSON.stringify(data));
          return onsuccess(data);
        }
      });
    };

    return UserAPI;

  })();

  module.exports = exports = UserAPI;

}).call(this);
