// spaces-client/lib/wiki.js
// Generated by CoffeeScript 1.6.3
(function() {
  var Wiki, exports;

  Wiki = (function() {
    var Faker, Resource;

    function Wiki() {}

    Faker = require('faker');

    Resource = require('./resource');

    Wiki.createWiki = function(site, sessionId, resourceId, parentId, userId, onsuccess, onfail, oncreate) {
      var data;
      data = {
        wiki: {
          name: Faker.Lorem.sentences(1),
          description: Faker.Lorem.sentences(4)
        }
      };
      if (parentId) {
        data['wiki']['parent_id'] = parentId;
      }
      return Resource.createResource(site, userId, sessionId, 'wikis', data, onsuccess, onfail, oncreate);
    };

    Wiki.showWiki = function(site, sessionId, resourceId, parentId, userId, onsuccess, onfail) {
      var addlHeaders;
      addlHeaders = {
        "Accept": "application/vnd.moxiesoft.spaces-v1+json",
        "Cookie": ["_social_navigator_session=" + sessionId, "path=/"]
      };
      return Resource.getResource(site, userId, sessionId, 'wikis/' + resourceId, addlHeaders, onsuccess, onfail);
    };

    Wiki.updateWiki = function(site, sessionId, resourceId, parentId, userId, onsuccess, onfail) {
      var data;
      data = {
        wiki: {
          description: Faker.Lorem.sentences(4)
        }
      };
      return Resource.editResource(site, userId, sessionId, 'wikis/' + resourceId, data, onsuccess, onfail);
    };

    Wiki.deleteWiki = function(site, sessionId, resourceId, parentId, userId, onsuccess, onfail, ondestroy) {
      return Resource.deleteResource(site, userId, sessionId, 'wikis/' + resourceId, onsuccess, onfail, ondestroy);
    };

    Wiki.createWikipage = function(site, sessionId, resourceId, parentId, userId, onsuccess, onfail, oncreate) {
      var data;
      data = {
        wiki_page: {
          title: Faker.Lorem.sentences(1),
          body: Faker.Lorem.sentences(4),
          parent_id: parentId
        }
      };
      return Resource.createResource(site, userId, sessionId, 'wiki_pages', data, onsuccess, onfail, oncreate);
    };

    Wiki.showWikipage = function(site, sessionId, resourceId, parentId, userId, onsuccess, onfail) {
      var addlHeaders;
      addlHeaders = {
        "Accept": "application/vnd.moxiesoft.spaces-v1+json",
        "Cookie": ["_social_navigator_session=" + sessionId, "path=/"]
      };
      return Resource.getResource(site, userId, sessionId, 'wiki_pages/' + resourceId, addlHeaders, onsuccess, onfail);
    };

    Wiki.updateWikipage = function(site, sessionId, resourceId, parentId, userId, onsuccess, onfail) {
      var data;
      data = {
        wiki_page: {
          body: Faker.Lorem.sentences(4)
        }
      };
      return Resource.editResource(site, userId, sessionId, 'wiki_pages/' + resourceId, data, onsuccess, onfail);
    };

    Wiki.deleteWikipage = function(site, sessionId, resourceId, parentId, userId, onsuccess, onfail, ondestroy) {
      return Resource.deleteResource(site, userId, sessionId, 'wiki_pages/' + resourceId, onsuccess, onfail, ondestroy);
    };

    return Wiki;

  })();

  module.exports = exports = Wiki;

}).call(this);
